src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -0,0 +1,621 @@
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.krad.bo.Note;
+
+import org.kuali.kfs.fp.document.DisbursementVoucherConstants;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.SupplierDiversity;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorContact;
+import org.kuali.kfs.vnd.businessobject.VendorContactPhoneNumber;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.kfs.vnd.businessobject.VendorSupplierDiversity;
+
+import edu.cornell.kfs.pmw.batch.PaymentWorksConstants;
+import edu.cornell.kfs.pmw.batch.PaymentWorksConstants.PaymentWorksTaxClassification;
+import edu.cornell.kfs.pmw.batch.businessobject.KfsVendorDataWrapper;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksIsoFipsCountryItem;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor;
+import edu.cornell.kfs.pmw.batch.report.PaymentWorksNewVendorRequestsBatchReportData;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksBatchUtilityService;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksVendorToKfsVendorDetailConversionService;
+import edu.cornell.kfs.vnd.CUVendorConstants;
+import edu.cornell.kfs.vnd.businessobject.CuVendorAddressExtension;
+import edu.cornell.kfs.vnd.businessobject.CuVendorSupplierDiversityExtension;
+import edu.cornell.kfs.vnd.businessobject.VendorDetailExtension;
+
+public class PaymentWorksVendorToKfsVendorDetailConversionServiceImpl implements PaymentWorksVendorToKfsVendorDetailConversionService {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.class);
+    
+    protected DateTimeService dateTimeService;
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
+            
+    @Override
+    public KfsVendorDataWrapper createKfsVendorDetailFromPmwVendor(PaymentWorksVendor pmwVendor,
+                                                                   Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap,
+                                                                   Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap) {
+        KfsVendorDataWrapper kfsVendorDataWrapper = createVendorDetail(pmwVendor, paymentWorksIsoToFipsCountryMap, paymentWorksToKfsDiversityMap);
+        return kfsVendorDataWrapper;
+    }
+
+    protected KfsVendorDataWrapper createVendorDetail(PaymentWorksVendor pmwVendor, 
+                                                      Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap, 
+                                                      Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap) {
+        //REMOVE THIS IF-STATEMENT WHEN FIRST PO IMPLEMENTATION IS DONE. 
```

Comment:
perhaps this should be removed altogether, or a method that does nothing could be added for the PO implementation. It would probably be good to have a separate explicit method for the Domestic DV vendors implementation?

Reply:
Per our offline discussion, PO vendors contain all of the data that comprise DV vendors plus more data. This if-statement is present to short circuit processing right away if the vendor is detected to be a PO vendor because PaymentWorks could send a PO vendor's data to this initial batch job implementation that is not equipped to handle it. I have added a more descriptively named method and removed the comments.